istioctl/pkg/waypoint/waypoint.go

```diff
@@ -133,8 +135,32 @@ func Cmd(ctx cli.Context) *cobra.Command {
 				return fmt.Errorf("failed to create Kubernetes client: %v", err)
 			}
 			gw := makeGateway(true)
-			if err != nil {
-				return fmt.Errorf("failed to create Kubernetes client: %v", err)
+			// Determine which address type to apply the waypoint to.
+			validAddressTypes := map[string]bool{
```

Comment:
I think the address type can also be in the `makeGateway`, which works for `istioctl x waypoint generate` as well?